csrc/scheduler/registry.cpp

```diff
@@ -969,23 +970,23 @@ SchedulerRuntimeInfo::SchedulerRuntimeInfo(
     if (input_tv != nullptr && !input_tv->isCpuScalar()) {
       const auto& metadata =
           expression_evaluator_->evaluate(IrBuilder::metadataExpr(input_tv));
-      const auto& alloc_sizes = metadata["alloc_size"].as<std::vector>();
-      const auto& alloc_strides = metadata["alloc_stride"].as<std::vector>();
+      const auto& alloc_sizes = metadata->*&TensorMetaData::alloc_size;
+      const auto& alloc_strides = metadata->*&TensorMetaData::alloc_stride;
```

Comment:
Can you just remind me again when to use `metadata->*` vs `metadata->`?

Reply:
We should always use `->*`. The way how the overloading of `->` in C++ works disallow us to do anything useful with it for supporting dynamic type.